In [20]:
import pandas as pd

In [21]:
df = pd.read_csv("Advertising.csv")

In [22]:
df.head()

,Unnamed: 0,TV,Radio,Newspaper,Sales
0,1,230.1,37.8,69.2,22.1
1,2,44.5,39.3,45.1,10.4
2,3,17.2,45.9,69.3,9.3
3,4,151.5,41.3,58.5,18.5
4,5,180.8,10.8,58.4,12.9


In [23]:
df = df.drop(columns=["Unnamed: 0"])

In [24]:
df.head()

,TV,Radio,Newspaper,Sales
0,230.1,37.8,69.2,22.1
1,44.5,39.3,45.1,10.4
2,17.2,45.9,69.3,9.3
3,151.5,41.3,58.5,18.5
4,180.8,10.8,58.4,12.9


In [25]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   TV         200 non-null    float64
 1   Radio      200 non-null    float64
 2   Newspaper  200 non-null    float64
 3   Sales      200 non-null    float64
dtypes: float64(4)
memory usage: 6.4 KB


In [26]:
df.describe()

,TV,Radio,Newspaper,Sales
count,200.000000,200.000000,200.000000,200.000000
mean,147.042500,23.264000,30.554000,14.022500
std,85.854236,14.846809,21.778621,5.217457
min,0.700000,0.000000,0.300000,1.600000
25%,74.375000,9.975000,12.750000,10.375000
50%,149.750000,22.900000,25.750000,12.900000
75%,218.825000,36.525000,45.100000,17.400000
max,296.400000,49.600000,114.000000,27.000000


In [27]:
df.corr()

,TV,Radio,Newspaper,Sales
TV,1.000000,0.054809,0.056648,0.782224
Radio,0.054809,1.000000,0.354104,0.576223
Newspaper,0.056648,0.354104,1.000000,0.228299
Sales,0.782224,0.576223,0.228299,1.000000


Exploratory Data Analysis (EDA)

The dataset contains 200 observations and 4 variables: TV, Radio, Newspaper, and Sales.

All variables are numeric (float) and there are no missing values, so the dataset is clean and ready for modeling.

From the descriptive statistics, TV advertising has the largest average spending, followed by Newspaper and Radio.

The correlation analysis shows that TV advertising has the strongest relationship with Sales (≈0.78), followed by Radio (≈0.58).
Newspaper advertising shows a weak correlation with Sales (≈0.23), suggesting it may have a smaller effect on predicting sales.

In [28]:
X = df[["TV", "Radio", "Newspaper"]]
y = df["Sales"]

In [29]:
from sklearn.model_selection import train_test_split

In [30]:
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.30, random_state=42
)

In [31]:
print(X_train.shape)
print(X_test.shape)

(140, 3)
(60, 3)


Train-Test Split

The dataset was split into training (70%) and testing (30%) subsets using train_test_split.
The training set contains 140 observations, while the test set contains 60 observations.
The training data will be used to build the regression model, and the test data will be used to evaluate its predictive performance.

In [32]:
from sklearn.linear_model import LinearRegression

In [33]:
model = LinearRegression()

model.fit(X_train, y_train)

LinearRegression()

In [34]:
model.intercept_

np.float64(2.7089490925159065)

In [35]:
model.coef_

array([0.04405928, 0.1992875 , 0.00688245])

Model Coefficients

The intercept and coefficients above represent the parameters of the linear regression model relating advertising expenditures (TV, Radio, Newspaper) to Sales. These values are used to construct the regression equation and interpret how each advertising channel affects predicted sales.

In [36]:
y_train_pred = model.predict(X_train)
y_test_pred = model.predict(X_test)

In [37]:
from sklearn.metrics import r2_score, mean_absolute_error, mean_squared_error
import numpy as np

In [38]:
train_r2 = r2_score(y_train, y_train_pred)
train_mae = mean_absolute_error(y_train, y_train_pred)
train_rmse = np.sqrt(mean_squared_error(y_train, y_train_pred))
train_mape = np.mean(np.abs((y_train - y_train_pred) / y_train)) * 100

# Test metrics
test_r2 = r2_score(y_test, y_test_pred)
test_mae = mean_absolute_error(y_test, y_test_pred)
test_rmse = np.sqrt(mean_squared_error(y_test, y_test_pred))
test_mape = np.mean(np.abs((y_test - y_test_pred) / y_test)) * 100

print("TRAIN METRICS")
print("R2:", train_r2)
print("MAE:", train_mae)
print("RMSE:", train_rmse)
print("MAPE:", train_mape)

print("\nTEST METRICS")
print("R2:", test_r2)
print("MAE:", test_mae)
print("RMSE:", test_rmse)
print("MAPE:", test_mape)

TRAIN METRICS
R2: 0.9055159502227753
MAE: 1.1581502948072526
RMSE: 1.574596830590544
MAPE: 13.583791147559104

TEST METRICS
R2: 0.8609466508230368
MAE: 1.5116692224549086
RMSE: 1.9485372043446387
MAPE: 16.298720118158336


Q1. Is the model a good model for making predictions?

Yes, the model performs well for prediction. The R² value for the test dataset is approximately 0.86, meaning the model explains about 86% of the variance in Sales. Additionally, the training and testing R² values are relatively close (0.91 vs 0.86), indicating that the model generalizes well and is not significantly overfitting. The error metrics (MAE ≈ 1.51 and RMSE ≈ 1.95) are relatively small compared to the range of Sales values, suggesting that the predictions are reasonably accurate.

Q2. How does each predictor variable affect Sales?                             

Regression Coefficients Interpretation

The regression model produced the following equation:

Sales=2.71+0.044(TV)+0.199(Radio)+0.0069(Newspaper)

The intercept (2.71) represents the predicted Sales when spending on TV, Radio, and Newspaper advertising is zero.

Holding the other variables constant:

A one-unit increase in TV advertising increases predicted Sales by approximately 0.044 units. TV advertising also showed the strongest correlation with Sales during the EDA, indicating that it has an important relationship with sales.

A one-unit increase in Radio advertising increases predicted Sales by approximately 0.199 units, indicating a strong effect on sales per unit increase.

A one-unit increase in Newspaper advertising increases predicted Sales by only 0.0069 units, suggesting that newspaper advertising has a very small influence on Sales in this dataset.

Q3. How much variance in Sales is explained by all three variables together?

The R² value for the test dataset is approximately 0.86, meaning that the predictors (TV, Radio, and Newspaper advertising) together explain about 86% of the variance in Sales.

Q4. What is the average margin of error (in dollars) in the predicted value of Sales?

The Mean Absolute Error (MAE) for the test dataset is approximately 1.51. This means that, on average, the predicted Sales values differ from the actual Sales values by about 1.51 units (or dollars).

Q5. What is the average % error in predicting sales?

The Mean Absolute Percentage Error (MAPE) for the test dataset is approximately 16.3%. This indicates that the predicted Sales values differ from the actual Sales values by about 16% on average.